inventory/s3/orc_utils.go

```diff
@@ -0,0 +1,125 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	gproto "github.com/golang/protobuf/proto" //nolint:staticcheck // orc lib uses old proto
+	"github.com/scritchley/orc/proto"
+	"github.com/treeverse/lakefs/logging"
+	"modernc.org/mathutil"
+)
+
+const (
+	maxPostScriptSize  = 256
+	orcInitialReadSize = 500
+)
+
+// getTailLength reads the ORC postscript from the given file, returning the full tail length.
+// The tail length equals (footer + metadata + postscript + 1) bytes.
+func getTailLength(filename string, size int64, logger logging.Logger) (int, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			logger.Errorf("failed to close orc file after reading tail. file=%s, err=%w", f.Name(), err)
```

Comment:
You only pass `logger` to be able to use `defer`, but this style makes you silence the error on `Close`.